### PR TITLE
Use semantic versioning

### PR DESCRIPTION
**Pull request by flaviojs**
Change the current version to "0.17.0-dev".

When possible, use the commit hash as the build metadata.

Only GameVersion.cc receives GAME_VERSION to avoid big rebuilds when the commit changes.

----

This was split from #1008

### CODE REVIEW

**Comment by flaviojs** [comment]
Adopting this means the version must be set to:
 * `0.(next minor).0-dev` in the master branch
 * `0.(minor).(patch)` in release tags
 * `0.(minor).(next patch)-dev` in the 0.(minor).x branches

When releasing `1.0.0` we must decide what is a breaking change.
The master branch version will be `(major).(next minor).0-dev` before a breaking change and `(next major).0.0-dev` with a breaking change.

Having a tag means the version is a prerelease, so `0.17.0-dev` comes before `0.17.0`.
We are at major version 0 so there are no promises, we can break stuff as much as we want.

File: src/game/GameVersion.cc [hunk 1]
```diff
@@ -1,10 +1,12 @@
 #include "GameVersion.h"
 
 #ifndef GAME_VERSION
-	#define GAME_VERSION "XXXXX"
+	#define GAME_VERSION XXXXX
```
Comment (lynxlynxlynx) [inline src/game/GameVersion.cc:4]: Can this still be reached?

Comment (flaviojs) [reply src/game/GameVersion.cc:4]: Same as before, it not reached if our build system is used.
Should I remove?

Comment (lynxlynxlynx) [reply src/game/GameVersion.cc:4]: I'd remove it, yes, since it's quite sane to expect people to use our build system. :)

Comment (flaviojs) [reply src/game/GameVersion.cc]: Done

**Pull request merged**